[Bug Report] Seg fault on new system

**Describe the bug**
A seg violation occurs on a system where the config file is yet to be created.

```
WARN[2021-06-02 22:19:48] config file not found. Assuming new system...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1185810]

goroutine 1 [running]:
github.com/stashapp/stash/pkg/plugin.(*Cache).RegisterGQLHandler(...)
        /home/still/stash/pkg/plugin/plugins.go:47
github.com/stashapp/stash/pkg/api.Start()
        /home/still/stash/pkg/api/server.go:187 +0xeb0
main.main()
        /home/still/stash/main.go:19 +0x3c
```

**To Reproduce**
Steps to reproduce the behavior:
1. Ensure `~/.stash/config.yml` does not exist
2. Start stash (latest `develop` build)
3. Crash

**Expected behavior**
...for it to not crash.

**Screenshots**
N/A

**Stash Version: (from Settings -> About):**
`stash version: v0.7.0-40-g1bb5de12 (1bb5de12) - 2021-06-02 22:11:23`

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Debian 11 (WSL)
 - Browser [e.g. chrome, safari] N/A
 - Version [e.g. 22] N/A

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6] N/A
 - OS: [e.g. iOS8.1]  N/A
 - Browser [e.g. stock browser, safari] N/A
 - Version [e.g. 22] N/A

**Additional context**
The build was compiled locally on the system.

--- a/pkg/api/resolver_mutation_plugin.go
+++ b/pkg/api/resolver_mutation_plugin.go
@@ -40,7 +40,7 @@ func (r *mutationResolver) RunPluginTask(ctx context.Context, pluginID string, t
 }
 
 func (r *mutationResolver) ReloadPlugins(ctx context.Context) (bool, error) {
-	err := manager.GetInstance().PluginCache.ReloadPlugins()
+	err := manager.GetInstance().PluginCache.LoadPlugins()
 	if err != nil {
 		logger.Errorf("Error reading plugin configs: %s", err.Error())
 	}

--- a/pkg/manager/manager.go
+++ b/pkg/manager/manager.go
@@ -69,6 +69,7 @@ func Initialize() *singleton {
 			Config:        cfg,
 			JobManager:    job.NewManager(),
 			DownloadStore: NewDownloadStore(),
+			PluginCache:   plugin.NewCache(cfg),
 
 			TxnManager: sqlite.NewTransactionManager(),
 
@@ -170,17 +171,6 @@ func initLog() {
 	logger.Init(config.GetLogFile(), config.GetLogOut(), config.GetLogLevel())
 }
 
-func initPluginCache() *plugin.Cache {
-	config := config.GetInstance()
-	ret, err := plugin.NewCache(config.GetPluginsPath())
-
-	if err != nil {
-		logger.Errorf("Error reading plugin configs: %s", err.Error())
-	}
-
-	return ret
-}
-
 // PostInit initialises the paths, caches and txnManager after the initial
 // configuration has been set. Should only be called if the configuration
 // is valid.
@@ -188,10 +178,13 @@ func (s *singleton) PostInit() error {
 	s.Config.SetInitialConfig()
 
 	s.Paths = paths.NewPaths(s.Config.GetGeneratedPath())
-	s.PluginCache = initPluginCache()
-	s.ScraperCache = instance.initScraperCache()
-
 	s.RefreshConfig()
+
+	if err := s.PluginCache.LoadPlugins(); err != nil {
+		logger.Errorf("Error reading plugin configs: %s", err.Error())
+	}
+
+	s.ScraperCache = instance.initScraperCache()
 
 	// clear the downloads and tmp directories
 	// #1021 - only clear these directories if the generated folder is non-empty

--- a/pkg/plugin/plugins.go
+++ b/pkg/plugin/plugins.go
@@ -14,44 +14,40 @@ import (
 	"path/filepath"
 
 	"github.com/stashapp/stash/pkg/logger"
+	"github.com/stashapp/stash/pkg/manager/config"
 	"github.com/stashapp/stash/pkg/models"
 	"github.com/stashapp/stash/pkg/plugin/common"
 )
 
 // Cache stores plugin details.
 type Cache struct {
-	path       string
+	config     *config.Instance
 	plugins    []Config
 	gqlHandler http.HandlerFunc
 }
 
-// NewCache returns a new Cache loading plugin configurations
-// from the provided plugin path. It returns an new instance and an error
-// if the plugin directory could not be loaded.
+// NewCache returns a new Cache.
 //
-// Plugins configurations are loaded from yml files in the provided plugin
-// directory and any subdirectories.
-func NewCache(pluginPath string) (*Cache, error) {
-	plugins, err := loadPlugins(pluginPath)
-	if err != nil {
-		return nil, err
-	}
-
+// Plugins configurations are loaded from yml files in the plugin
+// directory in the config and any subdirectories.
+//
+// Does not load plugins. Plugins will need to be
+// loaded explicitly using ReloadPlugins.
+func NewCache(config *config.Instance) *Cache {
 	return &Cache{
-		path:    pluginPath,
-		plugins: plugins,
-	}, nil
+		config: config,
+	}
 }
 
 func (c *Cache) RegisterGQLHandler(handler http.HandlerFunc) {
 	c.gqlHandler = handler
 }
 
-// ReloadPlugins clears the plugin cache and reloads from the plugin path.
+// LoadPlugins clears the plugin cache and loads from the plugin path.
 // In the event of an error during loading, the cache will be left empty.
-func (c *Cache) ReloadPlugins() error {
+func (c *Cache) LoadPlugins() error {
 	c.plugins = nil
-	plugins, err := loadPlugins(c.path)
+	plugins, err := loadPlugins(c.config.GetPluginsPath())
 	if err != nil {
 		return err
 	}

